src/Connector.sol

```diff
@@ -102,23 +128,120 @@ contract CentrifugeConnector {
+        }
+
+        if(userValues.maxMint >= _trancheTokensAmount) { // case: user has unclaimed trancheTokens in escrow -> more than redemption request
+            uint256 userTrancheTokenPrice = calcCustomTrancheTokenPrice( _user, _tranche);
+            uint256 assets = _trancheTokensAmount * userTrancheTokenPrice;
+            _decreaseDepositLimits(_user, _tranche, assets, _trancheTokensAmount);
+        } else {
+            uint transferAmount = _trancheTokensAmount - userValues.maxMint;
+            userValues.maxDeposit = 0;
+            userValues.maxMint = 0;
+
+            require(tranche.balanceOf(_user) >= _trancheTokensAmount, "CentrifugeConnector/insufficient-tranche-token-balance");
+            require(tranche.transferFrom(_user, address(escrow), transferAmount), "CentrifugeConnector/tranche-token-transfer-failed");
+        } 
+
+        gateway.increaseRedeemOrder(cfgTranche.poolId, cfgTranche.trancheId, _user, tranche.asset(), _trancheTokensAmount);
+    }
+    
+
+    // TODO: fix uint256 - uint128
+    function requestDeposit(address _tranche, uint _currencyAmount, address _user) connectorsActive poolActive(_tranche) public auth {
+        UserTrancheValues userValues = orderbook[_user][_tranche];
+        CFGTranche cfgTranche = cfgTranches[_tranche];
+        TrancheLike tranche = TrancheLike(_tranche);
+        ERC20Like currency = ERC20Like(TrancheLike(_tranche).asset());
+
+        require(_poolCurrencyCheck(tranche.asset(), cfgTranche.poolId), "CentrifugeConnector/currency-not-supported");
+        require(_trancheTokenCheck(cfgTranche.poolId, cfgTranche.trancheId, _user), "CentrifugeConnector/tranche-tokens-not-supported");
+
+        if (userValues.openRedeem > 0) { // cancel outstanding redeem orders 
+            gateway.decreaseRedeemOrder(cfgTranche.poolId, cfgTranche.trancheId, _user, tranche.asset(), userValues.openRedeem);
+        }
+        if(_currencyAmount == 0) { // case: user only wants to cancel outstanding redemptions
+            return; 
+        }
+        if(userValues.maxWithdraw >= _currencyAmount) { // case: user has some claimable fund in escrow -> funds > Deposit request
+            uint256 userTrancheTokenPrice = calcCustomTrancheTokenPrice( _user, _tranche);
+            uint256 trancheTokens = _currencyAmount / userTrancheTokenPrice;
+            _decreaseRedemptionLimits(_user, _tranche, _currencyAmount, trancheTokens);
+        } else {
+            uint transferAmount = _currencyAmount - userValues.maxWithdraw;
+            userValues.maxWithdraw = 0;
+            userValues.maxRedeem = 0;
+
+            require(currency.balanceOf(_user) >= transferAmount, "CentrifugeConnector/insufficient-balance");
+            require(currency.transferFrom(_user, address(escrow), transferAmount), "CentrifugeConnector/currency-transfer-failed");
+        } 
+
+        gateway.increaseInvestOrder(cfgTranche.poolId, cfgTranche.trancheId, _user, tranche.asset(), _currencyAmount);
```

Comment:
I think this line should be within the `else` statement, since there is no need to increase the invest order if `funds > Deposit request` is true